scrapy_poet/middleware.py

```diff
@@ -70,7 +86,7 @@ def process_request(self, request: Request, spider: Spider) -> Optional[DummyRes
         """
         if self.injector.is_scrapy_response_required(request):
             return None
-
+        yield from self._inject_cb_kwargs(request)
```

Comment:
Obviously it was added here for a reason, but wouldn't process_response be called anyways if DummyResponse is returned here?

Reply:
Hmm… That makes sense. I think I added that code while trying to figure out why things were not working, and once I figured out the actual reason (did not update `DEFAULT_PROVIDERS`) I did not realize this code was not necessary anymore.

Reply:
I'm asking because I'm not sure about the consequences of having it in process_request, maybe we'd get some dependencies too soon. E.g. there is a middleware which drops the request later, but we already called all the providers, which did e.g. external API calls.